Tests/ForageSDKTests/FloatingTextFieldTests.swift

```diff
@@ -0,0 +1,238 @@
+        XCTAssertNotNil(resultAttributedPlaceholder)
+        
+        let expectedAttributes: [NSAttributedString.Key: Any] = [.foregroundColor: placeholderColor]
+        let attributedString = NSAttributedString(string: placeholder, attributes: expectedAttributes)
+        XCTAssertEqual(resultAttributedPlaceholder, attributedString)
+    }
+    
+    func test_placeholderWithoutColor() {
+        let placeholder = "PAN number"
+        floatingTextField.placeholder = placeholder
+        floatingTextField.placeholderColor = nil
+        
+        XCTAssertNil(floatingTextField.placeholderColor)
+    }
+    
+    func test_textRect_insetRectForEmptyBounds() {
+        let x: CGFloat = 8
+        let paddingX: CGFloat = 8
+        let bounds = CGRect(x: 0, y: 0, width: 100, height: 100)
+        
+        let expectedRect = CGRect(x: x,
+                                  y: 0,
+                                  width: bounds.width - x - paddingX,
+                                  height: bounds.height)
+        let resultRect = floatingTextField.textRect(forBounds: bounds)
+        XCTAssertEqual(resultRect, expectedRect)
+    }
+    
+    func test_textRect_insetRectForBounds() {
+        let x: CGFloat = 8
+        let paddingX: CGFloat = 8
+        let paddingYFloatLabel: CGFloat = 5
+        let bounds = CGRect(x: 0, y: 0, width: 100, height: 100)
+        
+        floatingTextField.text = "0000 0000"
+        floatingTextField.placeholder = "PAN number"
+        
+        let topInset = paddingYFloatLabel + floatingTextField.bounds.size.height
+        let textOriginalY = floatingTextField.frame.height / 2.0
+        var textY = topInset - textOriginalY
+        
+        if textY < 0 { textY = topInset }
+        
+        let expectedRect = CGRect(x: x,
+                                  y: ceil(textY) - 2,
+                                  width: bounds.size.width - x - paddingX,
+                                  height: bounds.height - topInset)
+        
+        let resultRect = floatingTextField.textRect(forBounds: bounds)
+        XCTAssertEqual(resultRect, expectedRect)
+    }
+    
+    func test_editingRect_insetRectForBounds() {
+        let x: CGFloat = 8
+        let paddingX: CGFloat = 8
+        let paddingYFloatLabel: CGFloat = 5
+        let bounds = CGRect(x: 0, y: 0, width: 342, height: 37)
+        
+        floatingTextField.text = "1111"
+        floatingTextField.placeholder = "PAN number"
+        
+        let topInset = paddingYFloatLabel + floatingTextField.bounds.size.height
+        let textOriginalY = floatingTextField.frame.height / 2.0
+        var textY = topInset - textOriginalY
+        
+        if textY < 0 { textY = topInset }
+        
+        let expectedRect = CGRect(x: x,
+                                  y: ceil(textY) - 2,
+                                  width: bounds.size.width - x - paddingX,
+                                  height: bounds.height - topInset)
```

Comment:
This code appears copy/pasted from `private func insetRectForBounds` (inside of `FloatingTextField.swift`). It's important to remember that writing tests should meaningfully increase our confidence that the original code works as expected. If a test copies the implementation verbatim, then we are only gaining confidence that the Swift compiler works as expected; we are not increasing our confidence that the code we wrote works as we expect.

The way that we gain confidence that the code we wrote works as expected is by documenting very simple cases inside of unit tests that other developers will easily be able to understand. Here is an example (for illustration purposes only) of a test that is easy to read and that tests a single case. Noteably it does not copy the implementation details of the test and does not include if-statements or other calculations.
```swift
func test_editingRect_forGivenBounds() {
    // Setup
    let bounds = CGRect(x: 0, y: 0, width: 100, height: 50) // A simple 100x50 rect for clarity.

    // Let's assume that the FloatingTextField is designed such that when given these bounds and 
    // no text is set, the editing rect should be inset by 10 points from each side.
    let expectedRect = CGRect(x: 10, y: 10, width: 80, height: 30)

    // Act
    let resultRect = floatingTextField.editingRect(forBounds: bounds)

    // Assert
    XCTAssertEqual(resultRect, expectedRect)
}

```

Reply:
Updated. Thanks